bundles/framework/maplegend/Flyout.js

```diff
@@ -100,175 +105,47 @@ Oskari.clazz.define('Oskari.mapframework.bundle.maplegend.Flyout',
         refresh: function () {
             this._populateLayerList();
         },
+        showMetadataFlyout: function (event, uuid) {
+            event.stopPropagation();
+            this.sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
+                uuid: uuid
+            }]);
+        },
         /**
          * @method _populateLayerList
          * @private
          * @param {Object} layerListContainer reference to jQuery object representing the layerlist placeholder
          * Renders legend images as an accordion for the selected layers.
          */
         _populateLayerList: function () {
-            var me = this;
-            var cel = jQuery(this.container);
-            cel.empty();
-
-            var accordion = Oskari.clazz.create('Oskari.userinterface.component.Accordion');
-            accordion.insertTo(cel);
-
-            var sandbox = this.instance.getSandbox();
-
             // populate selected layer list
-            var layers = sandbox.findAllSelectedMapLayers().slice(0),
-                n,
-                layer,
-                layerContainer,
-                accordionPanel;
-
-            for (n = layers.length - 1; n >= 0; n -= 1) {
-                layer = layers[n];
-                layerContainer = this._createLayerContainer(layer);
+            const layers = this.sandbox.findAllSelectedMapLayers().slice(0);
+            const titles = [];
 
-                const layerTitle = jQuery('<div class="maplegend-layer-title">' + Oskari.util.sanitize(layer.getName()) + '</div>');
+            for (const layer of layers) {
                 const uuid = layer.getMetadataIdentifier();
-                if (uuid) {
-                    layerTitle.append(me.templateTools.clone());
-                    layerTitle.find('div.icon-info').on('click', function (event) {
-                        event.stopPropagation();
-                        sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
-                            uuid: uuid
-                        }]);
+                const layerLegendImage = layer.getLegendImage ? layer.getLegendImage() : null;
+
+                if (layerLegendImage) {
+                    titles.push({
+                        title: Oskari.util.sanitize(layer.getName()),
```

Comment:
The title is now rendered as part of JSX without `dangerouslySetInnerHTML` so we don't need to sanitize it here.